crates/core/src/db/queries/metadatas.rs

```diff
@@ -264,23 +269,38 @@ pub fn list(
         );
     }
 
-    if let Some(offerers) = offerers {
+    // ignoring the with_offers=false case for now, since it is more complicated and we dont currently have a need
+    let should_query_bid_receipts =
+        offerers.is_some() || (with_offers.is_some() && with_offers.unwrap_or(false));
+
+    if should_query_bid_receipts {
+        let mut bid_receipts_conditions = Condition::all().add(
+            Expr::tbl(BidReceipts::Table, BidReceipts::Metadata)
+                .equals(Metadatas::Table, Metadatas::Address),
+        );
+
+        if let Some(offerers) = offerers {
+            bid_receipts_conditions = bid_receipts_conditions
+                .add(Expr::col((BidReceipts::Table, BidReceipts::Buyer)).is_in(offerers))
+                .add(Expr::tbl(BidReceipts::Table, BidReceipts::PurchaseReceipt).is_null())
+                .add(Expr::tbl(BidReceipts::Table, BidReceipts::CanceledAt).is_null());
+        }
+
+        if let Some(with_offers) = with_offers {
+            if with_offers {
```

Comment:
Can't this just be `if let Some(true) = with_offers`?  If so then why's it an option?

Reply:
I didnt know about the `if let Some(true)` unpacking, pretty cool.

It's an option because `with_offers = false` implies you only want listings _without offers_ and that's not something I want to try to support right now. So, if you set it to `false` or dont provide it, then we ignore whether or not the NFT has offers, i.e. you'll get back NFTs with offers and without offers (versus only those with offers when `with_offers = true`)

Reply:
Hmm.  Should make a note or an error in the code for that if we don't support it, otherwise the API surface becomes confusing.

Reply:
It's already noted in the API documentation, but I will throw an error if it is `false` just to drive home the point :D